clang/test/SemaSYCL/float128.cpp

```diff
@@ -1,5 +1,5 @@
-// RUN: %clang_cc1 -triple spir64 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -verify -fsyntax-only %s
-// RUN: %clang_cc1 -triple x86_64-linux-gnu -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -fsyntax-only %s
+// RUN: %clang_cc1 -triple spir64 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -verify -fsyntax-only %s -verify-ignore-unexpected=warning
```

Comment:
Can you explain why you'd like to make this change to these tests? This seems like it'll lose significant test coverage, as any new warnings (or notes, in other tests) will not be caught by `-verify`.

Reply:
My reasonings were:
The original intent of this test wasn't to detect double literals in device code so adding that `expected-warning` to this test wasn't necessary.

The original test did not throw any warnings, so adding `-verify-ignore-unexpected=warning` wouldn't supress any warnings that were previously thrown by this test.

I also came across several other tests that used `-verify-ignore-unexpected=warning` to suppress warnings that were unrelated to what the test was testing ( one ex: llvm/sycl/test/basic_tests/atomic-ref-instantiation.cpp )

While this test requires only 1 expected-warning to be added, [supported_math.cpp](https://github.com/intel/llvm/blob/sycl/clang/test/SemaSYCL/supported_math.cpp) for example, has too many occurences of double literals in device code between lines 32 and 78. I'd had to add expected-warning for all those double literal occurences. 

I can certainly add the warings explicitly to each of the tests, but since I saw other tests which followed -`verify-ignore-unexpected=warning` standard, I figured it'd be okay.

Please let me know

Reply:
I would definitely *not* use that flag. I see it's being used in community as well, but some of those uses are pretty questionable, frankly (none of the uses in SemaCUDA seem like a good idea to me).

The better way to address this is:
* Tests which are specifically testing the behavior of `double` when run on device (so this warning is harmful), you can use `-Wno-double-in-device` to disable that warning entirely for the file.
* All other tests should be updated with the appropriate `expected-warning` lines.

I'd recommend forgetting that `-verify-ignore-unexpected` exists, that's how little it should be used in practice. :-)